artview/plugins/io.py

```diff
@@ -0,0 +1,201 @@
+    pyart.aux_io.read_radx]
+
+
+class AdvancedIO(Component):
+    '''
+    Open an interactive python console so the direct manipulation
+    '''
+
+    Vradar = None  #: see :ref:`shared_variable`
+    Vgrid = None  #: see :ref:`shared_variable`
+
+    @classmethod
+    def guiStart(self, parent=None):
+        '''Graphical interface for starting this class.'''
+        return self(), False
+
+    def __init__(self, dirIn=None, name="AdvancedIO", parent=None):
+        '''Initialize the class to create the interface.
+
+        Parameters
+        ----------
+        [Optional]
+        dirIn: string
+            Initial directory path to open.
+        name : string
+            Field Radiobutton window name.
+        parent : PyQt instance
+            Parent instance to associate to this class.
+            If None, then Qt owns, otherwise associated with parent PyQt
+            instance.
+        '''
+        super(AdvancedIO, self).__init__(name=name, parent=parent)
+        self.dirView = DirView(dirIn)
+        self.setCentralWidget(self.dirView)
+        self.dirView.opened.connect(self.open)
+        self.dirView.customContextMenuRequested.connect(self.contextMenu)
+        self.Vradar = Variable(None)
+        self.Vgrid = Variable(None)
+        self.sharedVariables = {"Vradar": None,
+                                "Vgrid": None}
+        self.connectAllVariables()
+        self.show()
+
+    def open(self, path):
+        ''' Slot of :py:attr:`DirView.opened` '''
+        # try several open
+        print ("open: %s" % path)
+        self.filename = str(path)
+        try:
+            radar = pyart.io.read(self.filename, delay_field_loading=True)
+            # Add the filename for Display
+            radar.filename = self.filename
+            self.Vradar.change(radar)
+            return
+        except:
+            try:
+                radar = pyart.io.read(self.filename)
+                # Add the filename for Display
+                radar.filename = self.filename
+                self.Vradar.change(radar)
+                return
+            except:
+                import traceback
+                print(traceback.format_exc())
+                radar_warning = True
+        try:
+            grid = pyart.io.read_grid(
+                self.filename, delay_field_loading=True)
+            self.Vgrid.change(grid)
+            return
+        except:
+            try:
+                grid = pyart.io.read_grid(self.filename)
+                self.Vgrid.change(grid)
+                return
+            except:
+                import traceback
+                print(traceback.format_exc())
+                grid_warning = True
+
+        if grid_warning or radar_warning:
+            msg = "Py-ART didn't recognize this file!"
+            common.ShowWarning(msg)
+        else:
+            msg = "Could not open file, invalid mode!"
+            common.ShowWarning(msg)
+        return
+
+    def contextMenu(self, pos):
```

Comment:
yes it is, it was a little trick to make it work, but this is exactly the place. The name "context menu" is also the standard name of this kind of functionality (case you want to google it up).